src/components/SearchBar/Autocomplete.tsx

```diff
@@ -98,8 +100,14 @@ export function Autocomplete(props) {
         ];
       },
       render({ children }, root) {
-        render(children, root);
-      },
+        if (!panelRootRef.current || rootRef.current !== root) {
+          rootRef.current = root;
+
+          panelRootRef.current?.unmount();
+          panelRootRef.current = createRoot(root);
+        }
+
+        panelRootRef.current.render(children);      },
```

Comment:
what is this doing? Is it possible to use `React.lazy` with Suspense here? https://legacy.reactjs.org/docs/code-splitting.html#reactlazy

or maybe `next/dynamic`? https://nextjs.org/docs/advanced-features/dynamic-import

Reply:
I'm also a little confused by this as well. Should we add some comments for why we're doing this?

Reply:
This is following the Algolia recommendation for upgrading to React18. Looking into the alternatives suggested.